app/src/main/java/org/septa/android/app/favorites/FavoritesFragment.java

```diff
@@ -198,8 +279,17 @@ private void refreshFavorite(final Favorite favorite, final NextToArriveTripView
         results.enqueue(new Callback<NextArrivalModelResponse>() {
             @Override
             public void onResponse(@NonNull Call<NextArrivalModelResponse> call, @NonNull Response<NextArrivalModelResponse> response) {
-                if (response != null && response.body() != null)
-                    tripView.setNextToArriveData(new NextArrivalModelResponseParser(response.body()));
+                if (response != null && response.body() != null) {
```

Comment:
This is the spot you should be checking the "success" property of NextArrivalModelResponse and make sure it has a value of "true".  If it is false then you have an error condition even if you didn't get a hard HTTP Error.